Source/Tests/LinuxSyscalls/Syscalls.h

```diff
@@ -349,6 +349,18 @@ uint64_t CloneHandler(FEXCore::Core::CpuStateFrame *Frame, FEX::HLE::clone3_args
     return flags;
   }
 
+
+template<typename T>
+static const bool HasSyscallError(T Result) {
+  // Most raw syscalls return errors using the negative range
```

Comment:
I find this comment difficult to follow, partly because different things are thrown together that don't make it clear how specifically it relates to the rather funky-looking implementation. It's also not clear why you're explaining glibc behavior or mentioning errno.

Maybe this could be restructured:
* An actual docstring at the top, explaining:
  * What are we actually checking? (the raw syscall error or a glibc error? both?)
  * When should this function *not* be used? (ioctl, apparently?)
* In the implementation:
  * Where does the constant 4096 come from?
  * How do glibc and raw syscalls differ?

If possible, we should also avoid the implicit "unsigned > signed" comparison, since that only works if the reader knows the obscure C integer promotion rules by heart. (Worst case is we could use `0xff...ff - 0x4095`?)

Reply:
Updated comments.